system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.hpp

```diff
@@ -301,12 +307,12 @@ class SubscriberTopicStatisticsNode : public system_metrics_collector::MetricsMe
   /**
    * Subscriber to listen to incoming messages on a topic
    */
-  typename rclcpp::Subscription<T>::SharedPtr subscription_;
+  std::vector<typename rclcpp::Subscription<T>::SharedPtr> subscriptions_;
```

Comment:
Is `typename` required here?
Asking just because I'm not quite sure for this context.

Reply:
Yes, the typename is required here.

Reply:
I'm more interested in the "why".

Reply:
Because its expecting a type here.

Reply:
So its this: https://en.cppreference.com/w/cpp/language/dependent_name ?

Reply:
yes, I also read this: https://docs.microsoft.com/en-us/cpp/cpp/typename?view=vs-2019